api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/art/LastRiskCategorizationDataEvaluator.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.reporting.data.converter.definition.evaluator.art;
+
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.kenyaemr.reporting.data.converter.definition.art.LastRiskCategorizationDataDefinition;
+import org.openmrs.module.reporting.data.person.EvaluatedPersonData;
+import org.openmrs.module.reporting.data.person.definition.PersonDataDefinition;
+import org.openmrs.module.reporting.data.person.evaluator.PersonDataEvaluator;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Evaluates LastRiskCategorizationDataDefinition
+ */
+@Handler(supports= LastRiskCategorizationDataDefinition.class, order=50)
+public class LastRiskCategorizationDataEvaluator implements PersonDataEvaluator {
+
+    @Autowired
+    private EvaluationService evaluationService;
+
+    public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
+        EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
+
+        String qry = "select ml.patient_id,\n" +
+			"       mid(max(concat(ml.date_created ,concat(ml.description ))),20) as risk_score\n" +
```

Comment:
Is there an idea of a visit in the ML risk score table? I am a bit worried about scores which are retrospectively generated.

Reply:
I have seen an evaluation date column. I think that is more ideal than the date created column

Reply:
@ojwanganto  :  mid(max(concat(ml.date_created ,NULLIF(concat(date(ml.evaluation_date)),'0000-00-00'))),20)
Rationale:
I pick the max evaluation_date where  date_created <= date(:endDate)

This is because we lack visit_date in this table